src/Raven.Client/Http/RequestExecutor.cs

```diff
@@ -1324,12 +1324,50 @@ private static void ValidateClientKeyUsages(X509Certificate2 certificate)
                 throw new InvalidOperationException("Client certificate " + certificate.FriendlyName + " must be defined with the following 'Enhanced Key Usage': Client Authentication (Oid 1.3.6.1.5.5.7.3.2)");
         }
 
-        private static Func<HttpRequestMessage, X509Certificate2, X509Chain, SslPolicyErrors, bool>[] _serverCertificateCustomValidationCallback = Array.Empty<Func<HttpRequestMessage, X509Certificate2, X509Chain, SslPolicyErrors, bool>>();
+        private static RemoteCertificateValidationCallback[] _serverCertificateCustomValidationCallback = Array.Empty<RemoteCertificateValidationCallback>();
         private static ConcurrentDictionary<WeakReference<Action>, object> _liveClients = new ConcurrentDictionary<WeakReference<Action>, object>();
         private Action _updateHttpHandlerDelegate;// we need this to hold a reference to the action as long as the executer is alive
         private static readonly ReaderWriterLockSlim _serverCallbackRWLock = new ReaderWriterLockSlim();
 
+        // HttpClient and ClientWebSocket use certificate validation callbacks with different signatures.
+        // We need this translator for backward compatibility to allow the user to supply any of the two signatures.
+        public class CallbackTranslator
```

Comment:
can be made private?

Reply:
No, we're also using it in DatabaseChanges